internal/controller/kustomization_controller.go

```diff
@@ -1108,3 +1119,23 @@ func (r *KustomizationReconciler) patch(ctx context.Context,
 
 	return nil
 }
+
+// getOriginRevision returns the origin revision of the source artifact,
+// or the empty string if it's not present, or if the artifact itself
+// is not present.
+func getOriginRevision(src sourcev1.Source) string {
+	a := src.GetArtifact()
+	if a == nil {
+		return ""
+	}
+
+	for _, key := range []string{
+		"org.opencontainers.image.revision",
```

Comment:
Let's add this to the API package as:

```go
const ArtifactOriginRevisionAnnotation = "org.opencontainers.image.revision"
```

And instead of a for loop, we just do `a.Metadata[api.ArtifactOriginRevisionAnnotation]`

Reply:
And keep the loop...

Reply:
What's point of having a loop over an array with a single entry?

Reply:
Also the annotation has nothing to do with SC, this is convention used in the CLI now in KC. SC does not look at annotations it just copies whatever is in the upstream artifact.

Reply:
> What's point of having a loop over an array with a single entry?

Communicating the possibility of more items being added to that array in the future. But I agree that, since this is an internal implementation detail that can change later, we can remove the loop for now.

> Also the annotation has nothing to do with SC, this is a convention used in the CLI now in KC. SC does not look at annotations, it just copies whatever is in the upstream artifact.

I don't think this constant belongs in KC's API. I agree that this is an API, but certainly not one that KC is implementing/providing. KC is just the consumer of this API. If source-controller is not the right place, maybe somewhere else is, e.g. pkg/apis/oci.

Reply:
We have it pkg/oci but I don’t want to import a huge package here just for one static var. Let’s set the static var in the controller internal pkg then.

Reply:
fixed, thanks